### ISSUE TITLE
Correct behavior for weights in chnk.flam.kernbyindex

### ISSUE THREAD

**Issue by mrachh** [fastalgorithms/chunkie#42]
@jghoskins @askhamwhat @danfortunato 

What should the correct behavior for weights in chnk.flam.kernbyindex? Currently it expects the user to provide weights properly concatenated according opdims of various kernels. I have a version, that just acts on weights of the chnkrs and then fixes the opdims stuff inside the subroutine. 

This will affect the implementation in chnk.flam.chunkerflam.

**Comment by danfortunato** [comment]
I think that the chunker object should store its own weights. Therefore the user should not have to pass any weights to chnk.flam.kernbyindex, and the weights stored in the chunker object will be used automatically. Then inside the subroutine, we can take the chunker-stored weights and expand them to have opdims compatible size.

**Comment by askhamwhat** [comment]
I think @danfortunato is right. Weights should be like normals in this sense. Ready to go. Probably won't be too hard to pull through, but chunkerfunc and chunkerpoly will have to be updated.

**Comment by jghoskins** [comment]
I agree - as much as possible weights should be hidden from the user.
